src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -136,7 +136,7 @@ public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNu
      * @param containerId The container ID.
      */
     public void stop(@Nonnull EnvVars launchEnv, @Nonnull String containerId) throws IOException, InterruptedException {
-        LaunchResult result = launch(launchEnv, false, "stop", containerId);
+        LaunchResult result = launch(launchEnv, false, "stop", "--time=1", containerId);
```

Comment:
So would `--time=0`. That is what I do *not* want to do. The intent is to allow the container to shut down gracefully.

The call to `rm` below ensures that it is stopped _and_ removed, whatever else happens.